### PR TITLE
Added variable to turn off fancify-symbol in Clojure mode

### PR DESCRIPTION
**Pull request by oppenlander**
I'm not a fan of the fancy/pretty symbols. I modeled the variable naming after the Colors Layer.

### CODE REVIEW

**Comment by pesterhazy** [comment]
Totally agree. I had literally implemented the exact same pull request two hours after you, @oppenlander.

In fact, I'd argue `fancify-symbols` should be opt-in (i.e. nil by default). Unicode symbols in .clj files do not conform to the principle of least surprise.

**Comment by pesterhazy** [comment]
I went ahead and created an alternate PR: https://github.com/syl20bnr/spacemacs/pull/499 which also includes a bit of documentation.

Feel free to ignore that PR if this one is perfered.

File: contrib/lang/clojure/config.el [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+;;
+;;; License: GPLv3
+
+;; ---------------------------------------------------------------------------
+;; Prefixes
+;; ---------------------------------------------------------------------------
+
+;; Variables
+
+(defvar clojure-enable-fancify-symbols t
```
Comment (syl20bnr) [inline contrib/lang/clojure/config.el]: omg.... this is exactly my comment on the other PR

**Comment by syl20bnr** [comment]
@oppenlander can you update your PR to match the documentation provided in #499 ?

**Comment by oppenlander** [comment]
@syl20bnr How do you feel about changing it to be opt-in?  
I only kept it opt-out because I didn't want to change the default behavior.

**Comment by syl20bnr** [comment]
@oppenlander I'm ok to set it to `t` by default

**Comment by oppenlander** [comment]
@syl20bnr It is already `t` by default. I was proposing setting it to `nil`

**Comment by syl20bnr** [comment]
Set it to nil by default then

The important thing is to have a matching documentation.

Check README of git layer and its layer variables  to know where to put the doc

**Comment by oppenlander** [comment]
Done. Updated to more closely match the git layer, and fancify symbols are now opt-in.

**Comment by syl20bnr** [comment]
Great ! Thank you :+1: 

Cherry-picked into develop branch.

**Pull request closed without merging**